File: android/app/src/main/java/com/guerra/WaterCounter/widgets/MMKVHelper.kt
package com.guerra.WaterCounter.widgets

import android.content.Context
import com.tencent.mmkv.MMKV
import java.text.SimpleDateFormat
import java.util.*

object MMKVHelper {
    private var mmkv: MMKV? = null

    fun initialize(context: Context) {
        if (mmkv == null) {
            MMKV.initialize(context)
            mmkv = MMKV.defaultMMKV()
        }
    }

    private fun dateKey(date: Date = Date()): String {
        val sdf = SimpleDateFormat("yyyy-MM-dd", Locale.US)
        return sdf.format(date)
    }

    private fun storageKey(key: String): String {
        return "intake:$key"
    }

    fun getIntake(context: Context, date: Date = Date()): Int {
        initialize(context)
        val key = dateKey(date)
        val storageKey = storageKey(key)
        val value = mmkv?.decodeString(storageKey) ?: "0"
        return value.toIntOrNull() ?: 0
    }

    fun setIntake(context: Context, value: Int, date: Date = Date()) {
        initialize(context)
        val key = dateKey(date)
        val storageKey = storageKey(key)
        mmkv?.encode(storageKey, value.toString())
    }

    fun addIntake(context: Context, amount: Int, date: Date = Date()) {
        val current = getIntake(context, date)
        setIntake(context, current + amount, date)
    }

    fun getGoal(context: Context): Int {
        initialize(context)
        val value = mmkv?.decodeString("settings:goal") ?: "2000"
        return value.toIntOrNull() ?: 2000
    }
}

File: ios/WaterCounterWidgets/MMKVHelper.swift
import Foundation
import MMKV

class MMKVHelper {
    static let shared = MMKVHelper()
    private let mmkv: MMKV?
    
    private init() {
        // Initialize MMKV with App Group
        MMKV.initialize(rootDir: nil, groupDir: "group.com.guerra.WaterCounter", logLevel: .none)
        mmkv = MMKV(mmapID: "mmkv.default", mode: .multiProcess)
    }
    
    func dateKey(_ date: Date = Date()) -> String {
        let calendar = Calendar.current
        let year = calendar.component(.year, from: date)
        let month = String(format: "%02d", calendar.component(.month, from: date))
        let day = String(format: "%02d", calendar.component(.day, from: date))
        return "\(year)-\(month)-\(day)"
    }
    
    private func storageKey(for key: String) -> String {
        return "intake:\(key)"
    }
    
    func getIntake(for date: Date = Date()) -> Int {
        let key = dateKey(date)
        let storageKey = storageKey(for: key)
        guard let value = mmkv?.string(forKey: storageKey) else { return 0 }
        return Int(value) ?? 0
    }
    
    func setIntake(_ value: Int, for date: Date = Date()) {
        let key = dateKey(date)
        let storageKey = storageKey(for: key)
        mmkv?.set(String(value), forKey: storageKey)
    }
    
    func addIntake(_ amount: Int, for date: Date = Date()) {
        let current = getIntake(for: date)
        setIntake(current + amount, for: date)
    }
    
    func getGoal() -> Int {
        guard let value = mmkv?.string(forKey: "settings:goal") else { return 2000 }
        return Int(value) ?? 2000
    }
}


